src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -716,7 +719,7 @@ protected override BoundNode VisitReturnStatementNoAdjust(BoundReturnStatement n
             }
 
             TypeSymbolWithAnnotations returnType = GetReturnType(compilation, _currentMethodOrLambda);
-            if ((object)returnType != null)
+            if ((object)returnType != null && expr.Kind != BoundKind.SuppressNullableWarningExpression)
```

Comment:
This suppresses nested nullability warnings as well as top-level nullability warnings. And it seems incorrect to handle `return x!;` here but not other cases such as `return (x!);`, `return (C?)x!;`, etc.